client/src/modules/Opportunities/components/NoConsentView/index.test.tsx

```diff
@@ -29,66 +29,125 @@ describe('NoConsentView', () => {
     const modalTitle = await screen.findByText(confirmationModalInfo.en.header);
 
     expect(modal).toBeInTheDocument();
-
-    // Check that we have all visible texts
     expect(modalTitle).toBeInTheDocument();
-    confirmationModalInfo.en.availableDataList.forEach(text => expect(screen.getByText(text)).toBeInTheDocument());
 
-    // Check that title tooltip works correctly
-    const titleTooltipIcon = await screen.findByTestId(confirmationModalInfo.ru.header);
+    // close modal
+    fireEvent.click(screen.getByRole('button', { name: 'Cancel' }));
+
+    await waitFor(() => {
```

Comment:
curious won't it be sufficient to `act` the click and then check the document. as of waiter would just make the test slower

Reply:
1. ![image](https://github.com/rolling-scopes/rsschool-app/assets/67139199/9f5c7e5d-39c7-4d5e-b5ff-345fd0f94d2c)
2. It brakes tests even if we ignore eslint

Reply:
![image](https://github.com/rolling-scopes/rsschool-app/assets/67139199/55190d32-d8c7-4517-9022-b122019f4fe8)
still brakes tests, I guess we need to wait for modal to close

Reply:
there's something ain't right on the test: if I won't close modal on any test, it brakes next one, modal stays in the DOM for some reason, even if I use `cleanup` in `beforeEach`. I managed to fix it only with `waitFor`.

Reply:
got it sorry didn't catch the problem initially.